src/prompt/primary.go

```diff
@@ -139,4 +119,15 @@ func (e *Engine) writePrimaryRPrompt() {
 	e.write(strings.Repeat(" ", space))
 	e.write(e.rprompt)
 	e.write(terminal.RestoreCursorPosition())
+
+	if e.cached {
+		return
+	}
+
+	e.updatePromptCache(&engineCache{
+		Prompt:            e.prompt.String(),
+		CurrentLineLength: e.currentLineLength,
+		RPrompt:           e.rprompt,
+		RPromptLength:     e.rpromptLength,
+	})
```

Comment:
How about leaving this part in the `Primary` method, and before its return?

```go
if !useCache {
  e.updatePromptCache(&engineCache{
    Prompt:            e.prompt.String(),
    CurrentLineLength: e.currentLineLength,
    RPrompt:           e.rprompt,
    RPromptLength:     e.rpromptLength,
  })
}

return e.string()
```

Reply:
Agreed, it's more explicit that way.